src/java.xml/share/classes/javax/xml/catalog/CatalogResolver.java

```diff
@@ -235,4 +235,53 @@ public InputStream resolveEntity(String publicId, String systemId,
     public LSInput resolveResource(String type, String namespaceUri,
             String publicId, String systemId, String baseUri);
 
+    /**
+     * Defines the actions that a CatalogResolver may take when it is unable to
+     * resolve an external reference. The actions are mapped to the string values
+     * of the {@link CatalogFeatures.Feature#RESOLVE RESOLVE} property.
+     *
+     * @since 22
+     */
+    public static enum NotFoundAction {
+        /**
+         * Indicates that the processing should continue as defined by the
+         * {@link CatalogFeatures.Feature#RESOLVE RESOLVE} property.
+         */
+        CONTINUE {
+            @Override
+            public String toString() { return "continue"; }
+        },
+        /**
+         * Indicates that the reference is skipped as defined by the
+         * {@link CatalogFeatures.Feature#RESOLVE RESOLVE} property.
+         */
+        IGNORE {
+            @Override
+            public String toString() { return "ignore"; }
+        },
+        /**
+         * Indicates that the resolver should throw a CatalogException as defined
+         * by the {@link CatalogFeatures.Feature#RESOLVE RESOLVE} property.
+         */
+        STRICT {
+            @Override
+            public String toString() { return "strict"; }
+        };
+
+        /**
+         * Returns the action type mapped to the specified
+         * {@link CatalogFeatures.Feature#RESOLVE resolve} property.
+         *
+         * @param resolve the value of the RESOLVE property
+         * @return the action type
+         */
+        static public NotFoundAction getType(String resolve) {
+            for (NotFoundAction type : NotFoundAction.values()) {
+                if (type.toString().equals(resolve)) {
+                    return type;
+                }
+            }
```

Comment:
Instead of looping through all the elements, `Enum.valueOf()` can be used (and it appropriately throws IAE if not found).

Reply:
Thanks. Added IAE. Enum.valueOf() is nice. However, in this case, the Enum type is mapped to the resolve property that was defined to be case sensitive. It would therefore require validating the case and then convert it to upper case before calling valueOf.

Reply:
That is correct. Thanks for checking